src/stacks-hierarchy/Root_AddPaymentMethodScreen/Root_AddPaymentMethodScreen.tsx

```diff
@@ -31,6 +37,56 @@ export const Root_AddPaymentMethodScreen = ({navigation}: Props) => {
     [navigation],
   );
 
+  useEffect(() => {
+    /**
+     * Listens to app state change, if app comes back from backgrounded/inactive
+     * state, it will then trigger the webview to load the {@link callbackUrl}, 
+     * which will redirect the user to the main payment card screen, and triggers
+     * a reload, so the user can see their added card(s). 
+     */
+    const subscription = AppState.addEventListener('change', (nextAppState) => {
```

Comment:
Because it is a listener, I think it could be nice to move out of the useEffect, and add as a dependency the app state to then inject the JS. What do you think?

Reply:
move out this part from `useEffect`?

```
const subscription = AppState.addEventListener('change', (nextAppState) => {
      if (appState.match(/inactive|background/) && nextAppState === 'active') {
        webViewRef.current?.injectJavaScript(callbackUrl);
      }
    });
```

Reply:
Yes, it's listening for events without need for a useEffect

Reply:
Hmm, should it though?

Since, if we create the const inside `useEffect`, then it will only be re-created when the `useEffect` is triggered, instead of every render. Also, many of the similar usages (the `addEventListener` part) in our code is created inside a `useEffect`. Moreover the official documents for `AppState` usage in https://reactnative.dev/docs/appstate, shows that in their sample code, the `const` part is created inside a `useEffect`. 

I'm not entirely sure, I think creating it inside the `useEffect` makes more sense because of the reason I mentioned above. Maybe @gorandalum could elaborate more on this? 😁

Reply:
All subscriptions/listeners need to be created inside a `useEffect`, or else they will be resubscribed every render.